plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/visitors/NoSelfChecker.java

```diff
@@ -159,7 +160,8 @@ public void beforeFunctionDef(FunctionDef node) {
                             new Tuple<Expected, FunctionDef>(new Expected("self or cls", received), node));
                 }
 
-            } else if (!startsWithSelf && !startsWithCls && !isStaticMethod && !isClassMethod) {
+            } else if (!startsWithSelf && !startsWithCls && !isStaticMethod && !isClassMethod
+                    && !hasInterfaceInBase(node)) {
```

Comment:
You shouldn't be checking this for each method, rather, compute the `hasInterfaceInBase` only once (it should probably be `hasZopeInterfaceInBase` as other libraries may also be `Interface`).

Also, it shouldn't just check for the string, rather it should find the definition... in `com.python.pydev.analysis.scopeanalysis.AbstractScopeAnalyzerVisitor.visitClassDef(ClassDef)`, when visiting the bases (when it's already finding the definition of the names) whether a given name will end up mapping to a zope interface and then store that info (as it's a visitor, you should push/pop that it's visiting bases and then on `markRead` if it's visiting bases mark whether it'll end up mapping to a zope interface if that's the case -- that should also be stored in a stack and popped appropriately when the class visit finishes).